apis/v1alpha1/db_cluster_parameter_group.go

```diff
@@ -88,20 +88,8 @@ type DBClusterParameterGroupSpec struct {
 	//
 	// This value is stored as a lowercase string.
 	// +kubebuilder:validation:Required
-	Name *string `json:"name"`
-	// A list of parameters in the DB cluster parameter group to modify.
-	//
-	// Valid Values (for the application method): immediate | pending-reboot
-	//
-	// You can use the immediate value with dynamic parameters only. You can use
-	// the pending-reboot value for both dynamic and static parameters.
-	//
-	// When the application method is immediate, changes to dynamic parameters are
-	// applied immediately to the DB clusters associated with the parameter group.
-	// When the application method is pending-reboot, changes to dynamic and static
-	// parameters are applied after a reboot without failover to the DB clusters
-	// associated with the parameter group.
-	Parameters []*Parameter `json:"parameters,omitempty"`
+	Name               *string            `json:"name"`
+	ParameterOverrides map[string]*string `json:"parameterOverrides,omitempty"`
```

Comment:
We 100% cannot remove, or modify the shape of, an existing spec field. It breaks backwards compatibility and would have pretty strange effects for existing customers. I understand this complicates the implementation, but if we can fit the changes within the existing spec that would be nice, otherwise we have to add a new field altogether.

Reply:
OK, just to be clear, is this only the rule for GA controllers?

Because an analogous change was made last September for `db_parameter_group` [here](https://github.com/aws-controllers-k8s/rds-controller/pull/107/files#diff-fbd5c921ed4d5f67af22264e781b19e636ba01ca7e8c8c95c043bfb8ccb3ecb2)

Reply:
>OK, just to be clear, is this only the rule for GA controllers?

Yes.

>Because an analogous change was made last September for db_parameter_group [here](https://github.com/aws-controllers-k8s/rds-controller/pull/107/files#diff-fbd5c921ed4d5f67af22264e781b19e636ba01ca7e8c8c95c043bfb8ccb3ecb2)

That change was made few days before going GA :/. If we don't have multi-version support we cannot delete/rename any fields.

Reply:
As discussed in the community meeting this morning, it is worth noting that this is a field that hasn't actually ever worked.  I'm a bit dubious that anyone might actually be using it.  Also, there is a value to having `db_parameter_group` and `db_cluster_parameter_group` behave similarly, since they're both effectively doing the same thing.

Reply:
Restored `Parameters` in `generator.yaml` and marked it as `DEPRECATED` using custom doc strings